subscription/src/main/java/org/killbill/billing/subscription/api/svcs/DefaultSubscriptionInternalApi.java

```diff
@@ -317,7 +328,13 @@ private boolean isStandaloneSpecifier(final Catalog catalog, final DateTime effe
                 subscriptionAndAddOns.add(subscriptionAndAddOnsSpecifier);
             }
 
-            return apiService.createPlansWithAddOns(accountId, subscriptionAndAddOns, catalog, callContext);
+            final List<SubscriptionBaseWithAddOns> subscriptionBaseWithAddOns = apiService.createPlansWithAddOns(accountId, subscriptionAndAddOns, catalog, callContext);
+            for (final SubscriptionBaseWithAddOns subscriptionBaseWithAO : subscriptionBaseWithAddOns) {
+                for (final SubscriptionBase subscriptionBase : subscriptionBaseWithAO.getSubscriptionBaseList()) {
+                    bundleIdCacheController.putIfAbsent(subscriptionBase.getId(), subscriptionBaseWithAO.getBundle().getId());
```

Comment:
Would it make sense to populate these caches at a lower level -- and  in a similar way we do `id` caches. For instance, we could overwrite a `DAO::create` -- similar to [account](https://github.com/killbill/killbill/blob/work-for-release-0.19.x/account/src/main/java/org/killbill/billing/account/dao/DefaultAccountDao.java#L98) -- and have this logic integrated here?

Reply:
I had actually looked at `DefaultAccountInternalApi` (where we handle BCD and immutable account caches) to plug it at a similar level.

I think the `create` call is a bit special, because it is shared and generic, while in this case, it's custom to subscription.

Reply:
Just to close the loop, but not so important: My point was that we could potentially have a custom `DefaultSubscriptionDao:create` (similarly to what we did for account) to encapsulate this logic.